src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,312 @@
+        reportWriterService.writeNewLines(1);
+    }
+
+    private void writeAgencySummaryToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Agency> agencyResultGrouping = processingResults.getAgencyResults();
+        List<RassBusinessObjectUpdateResult<Agency>> agencyResults = agencyResultGrouping.getObjectResults();
+        writeBOResultsSummaryToReport(agencyResults, Agency.class, true);
+    }
+
+    private void writeProposalSummaryToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Proposal> proposalResultGrouping = processingResults.getProposalResults();
+        List<RassBusinessObjectUpdateResult<Proposal>> proposalResults = proposalResultGrouping.getObjectResults();
+        writeBOResultsSummaryToReport(proposalResults, Proposal.class, false);
+    }
+
+    private void writeAwardSummaryToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Award> awardResultGrouping = processingResults.getAwardResults();
+        List<RassBusinessObjectUpdateResult<Award>> awardResults = awardResultGrouping.getObjectResults();
+        writeBOResultsSummaryToReport(awardResults, Award.class, true);
+    }
+
+    private <R extends PersistableBusinessObject> void writeBOResultsSummaryToReport(List<RassBusinessObjectUpdateResult<R>> results, Class businessObjectClass,
+            boolean shouldReportUpdates) {
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SUCCESS_NEW, businessObjectClass);
+        if (shouldReportUpdates) {
+            writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SUCCESS_EDIT, businessObjectClass);
+        }
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.ERROR, businessObjectClass);
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SKIPPED, businessObjectClass);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsSummaryToReportByResultCode(List<RassBusinessObjectUpdateResult<R>> results,
+            RassObjectUpdateResultCode resultCode, Class businessObjectClass) {
+        List<RassBusinessObjectUpdateResult<R>> filteredResults = results.stream().filter(result -> resultCode.equals(result.getResultCode()))
+                .collect(Collectors.toList());
+        writeSummaryLineToReport(filteredResults, resultCode, businessObjectClass);
+    }
+
+    private <R extends PersistableBusinessObject> void writeSummaryLineToReport(List<RassBusinessObjectUpdateResult<R>> filteresdResults,
```

Comment:
The results list variable has a minor typo that should be corrected. Sorry for not noticing it the first time.

Reply:
fixed